helpers.js

```diff
@@ -18,17 +23,54 @@ let getUserByEmail = function(email, users) {
 
 // filters the URLs for only that user
 let userURLs = function(user, urlDatabase) {
-  let userDatabase = { };
+  let userDatabase = {};
   for (let url in urlDatabase) {
     if (urlDatabase[url].userID === user) {
-      userDatabase[url] = {longURL: urlDatabase[url].longURL, userID: user};
+      userDatabase[url] = { longURL: urlDatabase[url].longURL, userID: user };
     }
   }
   return userDatabase;
 };
 
+const hashPassword = (password, saltRounds = 10) => {
+  let salt = bcrypt.genSaltSync(saltRounds);
+  return bcrypt.hashSync(password, salt);
+};
+
+const getEmailList = (stateUsers) => {
+  // returns an array with the VALUES (not KEYS) in stateUsers
+  const arrayOfUserObjects = Object.values(stateUsers);
+  // reduce is your best friend you can do pretty much everything with it
+  return arrayOfUserObjects.reduce(
+    (arrayToBeReturned, currentUserObject) => [...arrayToBeReturned, currentUserObject.email],
+    [],
+  );
+};
+
+const emailExists = (stateUsers, email) =>
+  // See if the email exists already in the users object
+  getEmailList(stateUsers).some((userEmail) => userEmail === email);
```

Comment:
Declarative, using `Array.some`

Reply:
Also see here how the arrow function returns explicitly here. No need for curly brackets as you don't have any other logic before returning.